lib/ui/painting/image_decoder_impeller.cc

```diff
@@ -33,6 +33,49 @@
 namespace flutter {
 
 namespace {
+
+#ifdef FML_OS_ANDROID
+class MallocDeviceBuffer : public impeller::DeviceBuffer {
+ public:
+  MallocDeviceBuffer(impeller::DeviceBufferDescriptor desc, uint8_t* data)
+      : impeller::DeviceBuffer(desc) {
+    data_ = data;
+  }
+
+  ~MallocDeviceBuffer() override { free(data_); }
+
+  bool SetLabel(const std::string& label) override { return true; }
+
+  bool SetLabel(const std::string& label, impeller::Range range) override {
+    return true;
+  }
+
+  void Flush() override {}
+
+  uint8_t* OnGetContents() const override { return data_; }
+
+  bool OnCopyHostBuffer(const uint8_t* source,
+                        impeller::Range source_range,
+                        size_t offset) override {
+    for (auto i = 0u; i < source_range.length; i++) {
+      data_[offset + i] = source[source_range.offset + i];
+    }
+    return true;
+  }
+
+ private:
+  uint8_t* data_;
+
+  FML_DISALLOW_COPY_AND_ASSIGN(MallocDeviceBuffer);
+};
+
+std::shared_ptr<MallocDeviceBuffer> CreateMallocDeviceBuffer(
+    impeller::DeviceBufferDescriptor desc) {
+  return std::make_shared<MallocDeviceBuffer>(
+      desc, static_cast<uint8_t*>(malloc(desc.size)));
```

Comment:
nit: since it's an error to `free` memory that wasn't malloc'd, it's prudent to put the malloc in the constructor.